lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -73,15 +78,28 @@ if Enum.all?(
     end
 
     @impl true
-    def handle_other({:new_tracks, tracks}, _ctx, state) do
+    def handle_other({:new_tracks, tracks}, ctx, state) do
       new_tracks = Map.new(tracks, &{&1.id, &1})
 
       subscriptions =
-        Enum.filter(tracks, fn track -> :raw in track.format end)
+        tracks
+        |> Enum.filter(fn track -> :raw in track.format end)
         |> Enum.map(fn track -> {track.id, :raw} end)
 
-      {{:ok, notify: {:subscribe, subscriptions}},
-       Map.update!(state, :tracks, &Map.merge(&1, new_tracks))}
+      {:endpoint, endpoint_id} = ctx.name
+      ref = make_ref()
+      send(state.rtc_engine, {:subscribe, subscriptions, self(), endpoint_id, ref})
+
+      receive do
+        {^ref, :ok} ->
+          {:ok, Map.update!(state, :tracks, &Map.merge(&1, new_tracks))}
+
+        {^ref, {:error, track_id, reason}} ->
+          raise "Subscription fails on track: #{track_id} because of #{reason}"
+      after
+        5000 ->
+          raise "Timeout subscribing on track in Engine with pid #{state.rtc_engine}"
+      end
```

Comment:
We need to create a new function in RTC.Engine for subscribing that will return what is in `receive` block.